### PR TITLE
Refactor to avoid using package private stuff in Brave

### PR DESCRIPTION
**Pull request by dsyer**
Add support for MySQL tracing (broken because of cycle)
Move static beans into private local methods

### CODE REVIEW

File: zipkin-java-server/src/main/java/io/zipkin/server/brave/ApiTracerConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+@Configuration
+@Import(MySqlTracerConfiguration.class)
+public class ApiTracerConfiguration extends WebMvcConfigurerAdapter {
+
+  @Value("${zipkin.collector.port:9410}")
+  int scribePort;
+
+  @Bean
+  Brave brave() {
+    return new Brave.Builder("zipkin-query").spanCollector(braveSpanCollector()).build();
```
Comment (adriancole) [inline zipkin-java-server/src/main/java/io/zipkin/server/brave/ApiTracerConfiguration.java]: so much nicer

File: zipkin-java-server/src/main/java/io/zipkin/server/brave/MySqlTracerConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.context.annotation.Configuration;
+
+import com.github.kristofa.brave.Brave;
+import com.github.kristofa.brave.mysql.MySQLStatementInterceptor;
+import com.mysql.jdbc.Driver;
+
+/**
+ * @author Dave Syer
```
Comment (adriancole) [inline zipkin-java-server/src/main/java/io/zipkin/server/brave/MySqlTracerConfiguration.java]: if you're up, can you scrub this, otherwise I can

Comment (dsyer) [inline zipkin-java-server/src/main/java/io/zipkin/server/brave/MySqlTracerConfiguration.java]: Oops. I need to remember to switch those off. Spring sources always have them.

**Comment by dsyer** [comment]
Last commit removed author tag.

**Pull request merged**